include/git2/common.h

```diff
@@ -59,6 +59,12 @@ typedef size_t size_t;
 #define GIT_WIN32 1
 #endif
 
+#if defined(__has_feature)
+#if __has_feature(address_sanitizer)
+#define GIT_ASAN 1
```

Comment:
Got it - yeah, I don't think that a public header is the best place for this.  I would suggest `src/common.h` if you do want to put this in a header file.

But my preference would be to put this in the `CMakeLists.txt` itself, which I think makes it a bit more discoverable.  (And there's some prior art there - the similar functionality of `DEBUG_POOL` and `MSVC_CRTDBG` just fiddles compiler flags.)

I would just put:

    ADD_DEFINITIONS(-D GIT_ASAN)

in the `IF (ASAN)` block.